File: src/main/kotlin/com/learning/gpt/controller/ChatGptController.java
package com.learning.gpt.controller;

import com.learning.gpt.dto.*;
import com.learning.gpt.service.ChatGptService;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/ai/gpt")
@RequiredArgsConstructor
public class ChatGptController {

    @Value("${gpt.endpoint.gpt-free}")
    private String endpointFree;

    @Value("${gpt.endpoint.gpt-charged}")
    private String endpointCharged;

    private final ChatGptService chatGPTService;

    @GetMapping("/chat")
    public ResponseEntity<GptResponseDto> chat(
            @NotNull @RequestParam("model") String model,
            @NotNull @RequestParam("prompt") String prompt){
        return ResponseEntity.ok().body(chatGPTService.chat(model, prompt, endpointCharged));
    }

    @GetMapping("/models")
    public ResponseEntity<GptModelsResponseDto> selectModelList() {
        return ResponseEntity.ok().body(chatGPTService.models());
    }
}

File: src/main/kotlin/com/learn/ai/chat/controller/ChatGptController.kt
package com.learn.ai.chat.controller

import com.learn.ai.chat.dto.ChatGptResponse
import com.learn.ai.chat.dto.PromptRequest
import com.learn.ai.chat.dto.GptModelsResponse
import com.learn.ai.chat.service.ChatGptService
import org.jetbrains.annotations.NotNull
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/ai/gpt")
class ChatGptController(
    private val chatGptService: ChatGptService,
) {

    @GetMapping("/models")
    fun getModels(): ResponseEntity<GptModelsResponse> {
        return ResponseEntity.ok().body(chatGptService.getModels())
    }

    @GetMapping("/chat")
    fun chat(@NotNull @RequestParam("model") model: String,
             @NotNull @RequestBody prompt: PromptRequest): ResponseEntity<ChatGptResponse> {
        return ResponseEntity.ok().body(chatGptService.chat(model, prompt))
    }
}